nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2440,6 +2447,7 @@ private DocumentModel getDirectAccessibleParent(DocumentRef docRef) {
             DocumentModel docModel = readModel(doc);
             Map<String, Serializable> options = new HashMap<>();
             options.put(systemProperty, value != null);
+            options.put(BINARY_TEXT_PREFIX + systemProperty, value);
```

Comment:
notifying after the doc.setSystemProp is not enough ? fulltext field can be huge and will be serialized in bulk events

Reply:
Hm good point

Reply:
The thought had crossed my mind. I thought that "inline" events will be omitted from event bundles.  Anyway, I will go back to the original fix and ask Florent why it doesn't work.